render/src/html/render.rs

```diff
@@ -0,0 +1,200 @@
+use crate::render::{Context, Renderer};
+
+use super::{
+    highlight::{self, DEFAULT_THEME},
+    Html, HtmlAttribute, HtmlAttributes, HtmlElement,
+};
+
+#[derive(Debug, Default)]
+pub struct HtmlRenderer {}
+
+impl Renderer<Html> for HtmlRenderer {
+    fn render_paragraph(
+        &mut self,
+        paragraph: &unimarkup_parser::elements::atomic::Paragraph,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_inlines(&paragraph.content, context)?;
+
+        Ok(Html::new_nested("p", HtmlAttributes::default(), inner))
+    }
+
+    fn render_heading(
+        &mut self,
+        heading: &unimarkup_parser::elements::atomic::Heading,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_inlines(&heading.content, context)?;
+        let tag = format!("h{}", u8::from(heading.level));
+
+        let attributes = HtmlAttributes::from(vec![HtmlAttribute {
+            name: "id".to_string(),
+            value: Some(heading.id.clone()),
+        }]);
+
+        Ok(Html::new_nested(&tag, attributes, inner))
+    }
+
+    fn render_verbatim_block(
+        &mut self,
+        verbatim: &unimarkup_parser::elements::enclosed::Verbatim,
+        _context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        // TODO: improve handling of attributes
+        // let attributes = serde_json::from_str::<VerbatimAttributes>(
+        //     &verbatim.attributes.as_ref().cloned().unwrap_or_default(),
+        // )
+        // .ok();
+
+        // let language = match attributes.as_ref() {
+        //     Some(attrs) => attrs.language.clone().unwrap_or(PLAIN_SYNTAX.to_string()),
+        //     None => PLAIN_SYNTAX.to_string(),
+        // };
+        let language = "auto";
+
+        let html = Html::new_with_body(HtmlElement {
+            name: "div".to_string(),
+            attributes: HtmlAttributes::default(),
+            content: Some(highlight::highlight_html_lines(
+                &verbatim.content,
+                language,
+                DEFAULT_THEME,
```

Comment:
Looking into `syntect`, I think we will need to write our own highlighting functionality.
Using `syntect`, we would need to set a theme during rendering (which we currently do), but there is no theme that is able to switch between light and dark styling. Since light/dark preference depends mostly on the client browser, we cannot use these themes. So there is also no reason to put the default theme into the context struct.

tldr: highlighting/theming must change. we need to write our own highlighter on top of parsing from `syntect`

Reply:
I think we might be able to use [`ClassedHTMLGenerator`](https://docs.rs/syntect/latest/syntect/html/struct.ClassedHTMLGenerator.html) from `syntect`.

They also provide the possibility to convert themes to CSS, so we might include light and dark media queries for one light and one dark theme.

Reply:
should I change it already in for this PR?

Reply:
If possible I would suggest implementing it as part of this PR. Is there option in the config to choose theme? If so, we could probably add the option to the context.

One suggestion regarding context, it would probably make the most sense to getter methods on `Context`, instead of exposing the fields of `Context` directly. This would make it possible to change `Context` in a backwards-compatible manner. 

For example: 

```rust
impl Context {
    pub fn code_block_style(&self) -> T { // or code_block_theme or whatever fits
        // Return whatever the T is
    }

    // ...
```